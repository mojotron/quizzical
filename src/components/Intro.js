import "../styles/Intro.css";

const Intro = (props) => {
  return (
    <div className="Intro">
      <h1 className="Intro__heading">Quizzical</h1>
      <p className="Intro__info">have fun answering trivia questions</p>
      <button type="button" className="btn" onClick={props.handleQuizRunning}>
        Start quiz
      </button>
    </div>
  );
};

export default Intro;
