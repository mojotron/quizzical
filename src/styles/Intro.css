.Intro {
  height: inherit;
  width: inherit;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.Intro__heading {
  color: var(--dark-blue);
  font-weight: 600;
  text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  font-family: "Karla", "Inter", monospace;
  font-style: normal;
  font-weight: 700;
  font-size: 3rem;
  line-height: 3.7rem;
  margin-bottom: 1.5rem;
}
.Intro__info {
  font-size: 1.6rem;
  line-height: 1.9rem;
  padding-bottom: 2rem;
  color: var(--dark-blue);
}
.Intro fieldset {
  border: 1px solid var(--dark-white);
  background-color: inherit;
  padding: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 0.5rem;
}

.Intro legend {
  position: relative;
  margin: 0 auto;
  color: var(--light-blue);
  font-size: 1.3rem;
  font-family: inherit;
  font-weight: 600;
  letter-spacing: 0.05rem;
}
.Intro label {
  font-family: inherit;
  padding-right: 0.5rem;
  font-size: 1.3rem;
  color: var(--light-blue);
}
.Intro select {
  font-family: inherit;
  border: none;
  background: none;
  border: 1px solid var(--light-blue);
  border-radius: 0.5rem;
  padding: 1rem 2rem;
  text-align: center;
  color: var(--dark-blue);
}
