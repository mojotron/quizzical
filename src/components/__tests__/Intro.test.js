import { render, screen } from "@testing-library/react";
import Intro from "../Intro";

const categoriesMock = [
  {
    id: 20,
    name: "Mythology",
  },
  {
    id: 21,
    name: "Sports",
  },
  {
    id: 23,
    name: "History",
  },
];

describe("Intro component", () => {
  test("renders heading", () => {
    render(
      <Intro
        categories={categoriesMock}
        numberOfQuestions={"5"}
        difficulty={"any"}
        category={"any"}
        handleQuizRunning={jest.fn()}
        handleChange={jest.fn()}
      />
    );
    expect(screen.getByRole("heading").textContent).toBe("Quizzical");
  });

  test("renders info paragraph", () => {
    render(
      <Intro
        categories={categoriesMock}
        numberOfQuestions={"5"}
        difficulty={"any"}
        category={"any"}
        handleQuizRunning={jest.fn()}
        handleChange={jest.fn()}
      />
    );
    expect(
      screen.getByText(/have fun answering trivia questions/i)
    ).toBeInTheDocument();
  });

  test("renders snapshot", () => {
    const { component } = render(
      <Intro
        categories={categoriesMock}
        numberOfQuestions={"5"}
        difficulty={"any"}
        category={"any"}
        handleQuizRunning={jest.fn()}
        handleChange={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("categories options", () => {
    render(
      <Intro
        categories={categoriesMock}
        numberOfQuestions={"5"}
        difficulty={"any"}
        category={"any"}
        handleQuizRunning={jest.fn()}
        handleChange={jest.fn()}
      />
    );
    const optionElements = screen.getAllByRole("option");
    expect(optionElements.length).toBe(4);
    expect(optionElements[0].textContent).toBe("Any");
    expect(optionElements[1].textContent).toBe("Mythology");
    expect(optionElements[2].textContent).toBe("Sports");
    expect(optionElements[3].textContent).toBe("History");
  });

  test("start quiz button", () => {
    render(
      <Intro
        categories={categoriesMock}
        numberOfQuestions={"5"}
        difficulty={"any"}
        category={"any"}
        handleQuizRunning={jest.fn()}
        handleChange={jest.fn()}
      />
    );
    expect(
      screen.getByRole("button", { name: "Start quiz" })
    ).toBeInTheDocument();
  });
});
