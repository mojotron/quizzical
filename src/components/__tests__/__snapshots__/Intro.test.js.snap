// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro component renders snapshot 1`] = `undefined`;
